mesonbuild/dependencies/pkgconfig.py

```diff
@@ -145,7 +145,7 @@ def get_env(environment: 'Environment', for_machine: MachineChoice,
         return env
 
     @staticmethod
-    def setup_env(env: T.MutableMapping[str, str], environment: 'Environment', for_machine: MachineChoice,
+    def setup_env(env: T.Union[T.Dict[str, str], os._Environ[str]], environment: 'Environment', for_machine: MachineChoice,
```

Comment:
relying on private members of the os module seems like a great way to end up with things broken using pypy or a new release of python. I see two way to not relying on something that we've been explicitly told not to rely on:

- switch from `map.copy()` to `copy.copy(map)`
- define our own interface type and use that instead

That would look something like:

```python
    _K = T.TypeVar('_K')
    _V = T.TypeVar('_V')

    class MutableCopyableMapping(T.MutableMapping[_K, _V]):
        def copy(self) -> MutableCopyableMapping[_K, _V]: ...
```

Reply:
Neither one works. e.g I get several dozen errors across various call sites saying:

```
has incompatible type "Dict[str, str]"; expected "MutableCopyableMapping[str, str]"
```

Meanwhile, using copy.copy() just gets me:

```
incompatible type "MutableMapping[str, str]"; expected "Dict[str, str]"
```

Reply:
Anyway, I do not think this is a legitimate concern. Please remember that this code ***cannot*** break on PyPy ***or*** on new versions of python...

... because this is only evaluated by mypy, not at runtime, and the only thing that matters to mypy is that typeshed's stubs provide a definition for `stdlib/os/__init__.pyi::_Environ`.

Even if this stops being the case one day, the only thing that will break is mypy.

Reply:
I agree it's not nice to poke into private members, but since that's only affecting mypy, I can live with it. Maybe just declare that type in a single place instead of repeating `T.Union[T.Dict[str, str], os._Environ[str]]`? IMHO we don't define enough types in Meson, that often produce seral lines long types annotation. In C/C++ I would do a lot more `typedef`.